### PR TITLE
Turn off admin emails for users with "no-moderation-emails" tag

### LINKED ISSUE #6282: For moderator accounts, add ability to turn off email at /settings
**Please describe the desired behavior.**

This feature request is part of improving the moderators workflow, and was written with @jywarren and @bronwen9. 

This feature request is to add the ability to turn of moderator emails per user at /settings. 

This feature should only be shown to /moderators and /admins. 

**Additional context (optional)**
Screen Shot 2019-09-12 at 4 58 58 PM

Thanks!

### PR DESCRIPTION
**Pull request by anthony-zhou**
Fixes #6282

Modified app/mailers/admin_mailer.rb to check for "no-moderation-emails" tag before sending emails.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by anthony-zhou** [comment]
@publiclab/reviewers The error I'm getting with `rails test test/unit` (and in Travis) occurs on the line where I call `mod_user.has_tag`. I thought mod_user would be some kind of user object, but it seems to be a string. Does anyone know how I can get the appropriate object to call `has_tag` on?

File: app/mailers/admin_mailer.rb [hunk 1]
```diff
@@ -10,7 +10,11 @@ def notify_node_moderators(node)
     @node = node
     @user = node.author
     @footer = feature('email-footer')
-    moderators = User.where(role: %w(moderator admin)).collect(&:email)
+    all_moderators = User.where(role: %w(moderator admin)).collect(&:email)
+    moderators = []
+    all_moderators.each do |mod_user|
+      moderators << mod_user unless mod_user.has_tag('no-moderation-emails')
```
Comment (cesswairimu) [inline app/mailers/admin_mailer.rb]: the `has_tag` method need to be called on the user object.

File: app/mailers/admin_mailer.rb [hunk 2]
```diff
@@ -10,7 +10,11 @@ def notify_node_moderators(node)
     @node = node
     @user = node.author
     @footer = feature('email-footer')
-    moderators = User.where(role: %w(moderator admin)).collect(&:email)
+    all_moderators = User.where(role: %w(moderator admin)).collect(&:email)
```
Comment (cesswairimu) [inline app/mailers/admin_mailer.rb]: change this to 
   ` all_moderators = User.where(role: %w(moderator admin))`

File: app/mailers/admin_mailer.rb (hunk 1 above)
Comment (cesswairimu) [inline app/mailers/admin_mailer.rb]: then this to  `moderators << mod_user.email unless mod_user.has_tag('no-moderation-emails')`

**Comment by anthony-zhou** [comment]
@cesswairimu Thanks for the pointers. I understand what's going on now, and the build succeeded. Is there anything else to do to complete this feature in the backend?

**Comment by jywarren** [comment]
Awesome! Thank you! and thanks Cess too!

**Pull request merged**